agent/agent.go

```diff
@@ -4068,6 +4071,10 @@ func (a *Agent) ReloadConfig(newCfg *config.RuntimeConfig) error {
 		MaxConnsPerClientIP: newCfg.HTTPMaxConnsPerClient,
 	})
 
+	if err := lib.CheckLimitsFromMaxConnsPerClient(newCfg.HTTPMaxConnsPerClient); err != nil {
+		return err
```

Comment:
If I am following the code properly it seems that the signal handler for reload calls `cmd.handleReload`, which calls `cmd.readConfig`->`Builder.BuildAndValidate()`->`Builder.Validate()`, before calling this `Agent.ReloadConfig()` method.

If that is correct then I do not think we need this call here. The value would already have been validated by the the call to `Builder.Validate()`.

Reply:
Yes, fixed in next commit